src/neo/Network/P2P/Payloads/TransactionAttributeCollection.cs

```diff
@@ -0,0 +1,134 @@
+using Neo.IO;
+using Neo.IO.Json;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Neo.Network.P2P.Payloads
+{
+    public class TransactionAttributeCollection : ISerializable, IEnumerable<KeyValuePair<TransactionAttributeUsage, List<TransactionAttribute>>>
+    {
+        /// <summary>
+        /// Maximum number of attributes that can be contained within a transaction
+        /// </summary>
+        public const int MaxTransactionAttributes = 16;
+
+        private readonly Dictionary<TransactionAttributeUsage, List<TransactionAttribute>> _entries;
+
+        public int Size =>
+            IO.Helper.GetVarSize(_entries.Count) +      // count
+            _entries.Count +                            // usages
+            _entries.Values.Sum(u => u.GetVarSize());   // entries
```

Comment:
It's time consuming, especially the `GetVarSize` method.

Reply:
`_entries.Values.Sum(u => u.Size)` ?

Reply:
It's a list